src/canvas/canvas.service.ts

```diff
@@ -194,6 +196,24 @@ export class CanvasService {
     });
   }
 
+  public async giveAccessByTag(command: GiveAccessByTagCommand) {
+    const user = await this.userRepository.findOne({
+      where: { id: command.userId },
+    });
+    if (!user) {
+      throw new NotFoundException();
+    }
+    const accessibleWithTags = await this.readByTags(command.tagIds, user.id);
+    accessibleWithTags.forEach((canvas) =>
+      command.personIds.forEach((personId) =>
+        this.giveAccess({
```

Comment:
Do we want to allow to share by tag `SHARED`? Such a tag has a special meaning in this app. This would be equivalent to generating public read-only link.
 
This should be discussed and decided if it's intended behaviour.

Reply:
I think, you're mixing-up sharing by public link with giving access (ACL). It's perfectly okay from the system's perspective to give access to all publicly shared canvases to someone.

Reply:
We can assume that this is just another way to share canvas. It can be handled on front